rama-pac/src/pac_parser/mod.rs

```diff
@@ -0,0 +1,90 @@
+use boa_engine::{Context as JsContext, JsString, JsValue, Source, js_string, property::Attribute};
+use boa_runtime::Console;
+use rama_core::error::BoxError;
+use rama_http::Uri;
+
+const PAC_STDLIB: &str = r#"
+// spec-ish helpers
+function dnsDomainIs(host, domain) {
+    return host.length >= domain.length &&
+           host.substring(host.length - domain.length) === domain;
+}
+function shExpMatch(str, shexp) {
+    // very small glob impl: * only
+    var re = new RegExp('^' + shexp.replace(/[.+?^${}()|[\]\\]/g, '\\$&').replace(/\*/g, '.*') + '$');
+    return re.test(str);
+}
+function isPlainHostName(host) { return host.indexOf('.') < 0; }
+function localHostOrDomainIs(host, hostdom) { return host === hostdom || hostdom.startsWith(host + "."); }
+function dnsResolve(host) { return host; }          // stub; swap for real resolver if needed
+function myIpAddress() { return "127.0.0.1"; }      // stub
+function isInNet(ip, pattern, mask) { return false; } // stub
+"#;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ProxyDirective {
+    Direct,
+    Proxy(String),
```

Comment:
We can probably do better, type-wise, no? As in a SocketAddress, or something like that?